### DISCUSSION TITLE
<NuxtPage name="some-name" /> not working

### DISCUSSION THREAD

**Discussion by victorlmneves** [nuxt/nuxt#26844] [Questions]
Hi all
I'm migrating a Vue 3 project to Nuxt 3 and I'm currently facing a problem with making <NuxtPage name="some-name" /> work

Here's a reproduction where you can what I'm trying to achieve.
From what I saw on the docs this is the right way, but not sure if I need to update something on the route structure

https://stackblitz.com/edit/nuxt-starter-z7j8fy?file=app%2Frouter.options.js

Thanks

**Comment by IsraelOrtuno** [comment]
You don't need to define the routes manually. Nuxt uses file-system routes:
https://nuxt.com/docs/getting-started/routing

If for some reason you want to name each of your routes you could use `definePageMeta` on each route file:
https://nuxt.com/docs/api/utils/define-page-meta

**Reply by victorlmneves** [reply]
@IsraelOrtuno 
For my project I need, I only have a single page file, that is responsible for rendering all pages/routes and child/nested routes/pages
